tinygrad/renderer/cuda.py

```diff
@@ -0,0 +1,28 @@
+import functools
+from tinygrad.renderer.cstyle import uops_to_cstyle, CStyleLanguage
+from tinygrad.helpers import dtypes
+
+class CUDALanguage(CStyleLanguage):
+  kernel_prefix = "__global__ "
+  smem_prefix = "__shared__ "
+  smem_prefix_for_cast=False
+  arg_int_prefix = "const int"
+  barrier = "__syncthreads();"
+  float4 = "make_float4"
```

Comment:
If you want to refactor, you can't have make_float2 and make_float4 hit different paths.

Reply:
done